### DISCUSSION TITLE
Disable Global Search Field

### DISCUSSION THREAD

**Discussion by grafst** [filamentphp/filament#6762] [Ideas]
For Apps that don't use search or for apps that use the [spotlight](https://filamentphp.com/plugins/pxlrbt-spotlight) it would be great to have an option to disable the global search field
<img width="1407" alt="2023-06-13_15-03-01" src="https://github.com/filamentphp/filament/assets/8471055/510c4b2e-92b9-47af-bc9d-3588acf1d7ed">

**Comment by pboivin** [comment]
Hi @grafts, I agree a config option would be nice. For a quick workaround, you could publish just this view from core and make it empty or hide the search input :
filament/filament/resources/views/components/global-search/index.blade.php

**Reply by grafst** [reply]
thanks.
Do you know how to publish a single view?

**Reply by pboivin** [reply]
Yes, if you're on Mac/Linux/WSL, something like this should work:
```
mkdir -p resources/views/vendor/filament/components/global-search
cp vendor/filament/filament/resources/views/components/global-search/index.blade.php resources/views/vendor/filament/components/global-search
```

(basically just copying the file manually)

**Reply by grafst** [reply]
thank you so much, it worked flawlessly!
so that would work with the views of any other package aswell? 
basically copy the whole path but add a  'vendor/<package>' inside the views folder?

**Reply by pboivin** [reply]
You're welcome! Yes this will work for pretty much all 3rd party Blade views and can be really useful for custom integrations BUT, I think it should be a last resort kind of thing. Here's a really good overview of the problems with overriding Blade views:
https://filamentphp.com/blog/publishing-views-in-laravel

**Reply by grafst** [reply]
yes I agree, only as a last resort. In this case hiding the search shouldn't impact anything else to much.

**Reply by zeeshantariq08** [reply]
In PageResource:
set protected static ?string $recordTitleAttribute = 'title'; value to null

**Comment by danharrin** [comment]
The feature is in v3 👍

**Reply by EntroSolutions** [reply]
I cant find anything in the documentation, I cant disable it.

Published files dont have folder global-search

**Reply by dave-mills** [reply]
> I cant find anything in the documentation, I cant disable it.
> 
> Published files dont have folder global-search

You can disable global search per panel: https://filamentphp.com/docs/3.x/panels/resources/global-search#disabling-global-search

I think it will also hide automatically if none of your Resource classes have a $recordTitleAttribute set.

**Comment by EntroSolutions** [comment]
So there is not global setting I need to go and add this to all my resources?

**Reply by dave-mills** [reply]
The setting to disable global search is per-panel, not per-resource (docs [here](https://filamentphp.com/docs/3.x/panels/resources/global-search#disabling-global-search))

If you want to have global search enabled, you should add a `$recordTitleAttribute` property to each resource that should be included in the global search.
